src/overlays/actors/ovl_En_Zl1/z_en_zl1.c

```diff
@@ -27,39 +37,626 @@ const ActorInit En_Zl1_InitVars = {
+void EnZl1_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnZl1* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4B834.s")
+    if ((this->actionFunc != func_80B4B8B4) && (this->actionFunc != func_80B4BC78)) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    }
+    func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 5);
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc != func_80B4B8B4) {
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.x, this->actor.posRot.rot.x, 0xA, 0x3E8, 1);
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.posRot.rot.y, 0xA, 0x3E8, 1);
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.z, this->actor.posRot.rot.z, 0xA, 0x3E8, 1);
+    func_80B4AE18(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4B874.s")
+s32 func_80B4C340(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnZl1* this = (EnZl1*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4B8B4.s")
+    if ((limbIndex == 4) || (limbIndex == 3) || (limbIndex == 6) || (limbIndex == 5)) {
+        *dList = NULL;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4BBC4.s")
+    if (limbIndex != 10) {
+        if (limbIndex == 17) {
+            rot->x += this->unk_200.y;
+            rot->y += this->unk_200.z;
+            rot->z += this->unk_200.x;
+        }
+    } else {
+        rot->x += this->unk_206.y;
+        rot->y += this->unk_206.x;
+        rot->z += this->unk_206.z;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4BC78.s")
+void func_80B4C400(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* this) {
+    Vec3f vec = D_80B4E714;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4BF2C.s")
+    if (limbIndex == 17) {
+        Matrix_MultVec3f(&vec, &this->posRot2.pos);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/EnZl1_Update.s")
+void EnZl1_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnZl1* this = THIS;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4C340.s")
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_girlB.c", 2011);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl1/func_80B4C400.s")
+    gSPSegment(gfxCtx->polyOpa.p++, 0x08, SEGMENTED_TO_VIRTUAL(this->unk_1F4));
+    gSPSegment(gfxCtx->polyOpa.p++, 0x09, SEGMENTED_TO_VIRTUAL(this->unk_1F8));
+    gSPSegment(gfxCtx->polyOpa.p++, 0x0A, SEGMENTED_TO_VIRTUAL(this->unk_1EC));
```

Comment:
These 3 actor members should probably be made pointers (I guess UNK_PTR for now) instead of u32.